File: branches/5.x/Templates/Examples/BasicSamples/CSharp/CodeBehind.cst.cs
using System.ComponentModel;
using CodeSmith.Engine;

public class SampleCodeBehindClass : CodeTemplate
{
	private bool _propertyFromCodeBehind = false;
	
	[Category("Options")]
	[Description("This property is inherited from the code behind file.")]
	public bool PropertyFromCodeBehind
	{ 
		get {return _propertyFromCodeBehind;}
		set {_propertyFromCodeBehind = value;}
	}
	
	public string GetSomething()
	{
		return "Something";
	}
}

File: branches/5.x/Templates/Examples/BasicSamples/VisualBasic/CodeBehind.cst.vb
Imports System.ComponentModel
Imports CodeSmith.Engine

Public Class SampleCodeBehindClass
	Inherits CodeTemplate
	Private _propertyFromCodeBehind As Boolean = False

	<Category("Options")> _
	<Description("This property is inherited from the code behind file.")> _
	Public Property PropertyFromCodeBehind() As Boolean
		Get
			Return _propertyFromCodeBehind
		End Get
		Set
			_propertyFromCodeBehind = value
		End Set
	End Property

	Public Function GetSomething() As String
		Return "Something"
	End Function
End Class
